lib/tests/api/live.rs

```diff
@@ -421,6 +426,88 @@ async fn live_select_query() {
 	drop(permit);
 }
 
+#[tokio::test]
+async fn check_table_exists_for_live_query() {
+	let dbs = new_ds().await.unwrap().with_notifications();
+	let (ns, db, tb) = ("test", "test", "person");
+	let ses = Session::owner().with_ns(ns).with_db(db).with_rt(true);
+
+	// Initiate a live query statement
+	let lq_stmt = format!("LIVE SELECT * FROM {}", tb);
+	let live_query_response = &mut dbs.execute(&lq_stmt, &ses, None).await.unwrap();
+
+	let live_id = live_query_response.remove(0).result.unwrap();
+	let live_id = match live_id {
+		Value::Uuid(id) => id,
+		_ => panic!("expected uuid"),
+	};
+
+	// Verify that the table definition has been created.
+	let tx = dbs.transaction(Write, Optimistic).await.unwrap();
+	let table_occurrences= &*(tx.all_tb(ns, db).await.unwrap());
+	assert_eq!(table_occurrences.len(), 1);
+	assert_eq!(table_occurrences[0].name.0, tb);
+	tx.cancel().await.unwrap();
+
+	// Initiate a Create record
+	let create_statement = format!("CREATE {}:test_true SET condition = true", tb);
+	let create_response = &mut dbs.execute(&create_statement, &ses, None).await.unwrap();
+	assert_eq!(create_response.len(), 1);
+	let expected_record = Value::parse(
+		&format!(
+			"[{{
+				id: {}:test_true,
+				condition: true,
+			}}]",
+			tb
+		),
+	);
+	let tmp = create_response.remove(0).result.unwrap();
+	assert_eq!(tmp, expected_record);
+
+	// Create a new transaction to verify that the same table was used.
+	let tx = dbs.transaction(Write, Optimistic).await.unwrap();
+	let table_occurrences= &*(tx.all_tb(ns, db).await.unwrap());
+	assert_eq!(table_occurrences.len(), 1);
+	assert_eq!(table_occurrences[0].name.0, tb);
+	tx.cancel().await.unwrap();
+
+	// Validate notification
+	let notifications = dbs.notifications().expect("expected notifications");
+	let notification = notifications.recv().await.unwrap();
+	assert_eq!(
+		notification,
+		surrealdb_core::dbs::Notification::new(
+			live_id,
+			surrealdb_core::dbs::Action::Create,
+			Value::parse(
+				&format!(
+					"{{
+						id: {}:test_true,
+						condition: true,
+					}}",
+					tb
+				),
+			),
+		)
+	);
+}
+
+pub async fn new_ds() -> Result<Datastore, surrealdb_core::err::Error> {
```

Comment:
I can't seem to find a better way to reuse these functions and methods defined in the `helpers.rs` and `parse.rs`. 
Any better alternative here?

Reply:
Yes, you should use the `new_db` function defined on line 20/25 of this file :D The datastore is internal and these are external tests that go through the driver, so the `new_db` connects to the correct datastore via the rust driver. It provides an interface that still allows statements to be executed as raw test.

However, since this part of the code is used for external tests, we shouldn't be verifying internal behaviour here. The tests should pass irrespective of whether a table definition exists or not here (like, it could be artificial, or lazy, or abscent else in implementation). So this particular test needs to go to a new location.

I would recommend doing it inside the `kvs/tests` directory somewhere. Its a bit of a mess there, but it is likely a good enough granularity. Alternatively, somewhere that is associated with live query tests would be good.

I just had a search and see those previous tests got removed, but you should be able to do a new one if you have a look at a previous version of the codebase (2+ weeks ago).